access_test.go

```diff
@@ -272,6 +272,93 @@ func TestExtraScopes(t *testing.T) {
 
 }
 
+func TestPlatformToken(t *testing.T) {
+	sconfig := NewServerConfig()
+	sconfig.AllowedAccessTypes = AllowedAccessType{PLATFORM}
+	server := NewServer(sconfig, NewTestingStorage())
+	server.AccessTokenGen = &TestingAccessTokenGen{}
+	resp := server.NewResponse()
+
+	req, err := http.NewRequest("POST", "http://localhost:14000/appauth", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.SetBasicAuth("1234", "aabbccdd")
+
+	req.Form = make(url.Values)
+	req.Form.Set("grant_type", string(PLATFORM))
+	req.Form.Set("client_id", "1234")
+	req.PostForm = make(url.Values)
+
+	if ar := server.HandleAccessRequest(resp, req); ar != nil {
+		ar.Authorized=true
+		server.FinishAccessRequest(resp, req, ar)
+	}
+
+	if resp.IsError && resp.InternalError != nil {
+		t.Fatalf("Error in response: %s", resp.InternalError)
+	}
+
+	if resp.IsError {
+		t.Fatalf("Should not be an error")
+	}
+
+	if resp.Type != DATA {
+		t.Fatalf("Response should be data")
+	}
+
+	if d := resp.Output["access_token"]; d != "1" {
+		t.Fatalf("Unexpected access token: %s", d)
+	}
+
+	if d := resp.Output["refresh_token"]; d != "r1" {
+		t.Fatalf("Unexpected refresh token: %s", d)
+	}
+}
+
+func TestPlatformTokenWithoutClientSecret(t *testing.T) {
```

Comment:
suggested scenario : 
- basic header and client id on form parameter are absent 
- basic header preset but no client id 
- use of non existent client

Reply:
for the 3rd scenario, do you mean to write test case using invalid client? @rjjatson 
if that is the case, we will test it in iam service. because client validation is not implemetated here

Reply:
for the 3rd scenario im thinking to hit the case when the client is nil, access.go:517

Reply:
I see, if that the case we can do that by checking in the testing storage. Thanks
Updating

Reply:
updated